### PR TITLE
Add support for `virtualization.hypervisor` hardware requirement

### PR DESCRIPTION
**Pull request by skycastlelily**
Pull Request Checklist

* [x] implement the feature
* [x] extend the test coverage
* [x] mention the version
* [x] include a release note

### CODE REVIEW

**Comment by thrix** [comment]
@skycastlelily  seems this should be updated after #2734 and update the tests?

**Comment by skycastlelily** [comment]
Hi @Miroslav Vadkerti ***@***.***>  yes:)

@Milos Prchlik ***@***.***> You are so efficient^^

**Comment by happz** [comment]
@skycastlelily hello, please, rebase & add a simple unit test similar to https://github.com/teemtee/tmt/blob/main/tests/unit/provision/mrack/test_hw.py#L452? This PR is fairly simple, it should be easy to get it reviewed and merged quickly.

**Comment by skycastlelily** [comment]
Really sorry for the late reply, I'm just back from my national holiday.

> This PR is fairly simple, This PR is fairly simple, it should be easy to
get it reviewed and merged quickly.

Yes ,and updated^^

**Comment by happz** [comment]
Sorry for not mentioning it in the previous post, but a short release note, similar to https://github.com/teemtee/tmt/blob/main/docs/releases.rst?plain=1#L44, would be nice. No more comments from my side.

**Comment by skycastlelily** [comment]
updated^^

**Review by therazix — APPROVED** [review]

File: spec/hardware/virtualization.fmf [hunk 1]
```diff
@@ -38,4 +38,4 @@ example:
 link:
   - implemented-by: /tmt/steps/provision/artemis.py
   - implemented-by: /tmt/steps/provision/mrack.py
-    note: "``virtualization.is-virtualized`` only"
+    note: "``virtualization.is-virtualized`` ``virtualization.hypervisor`` only"
```
Comment (happz) [inline spec/hardware/virtualization.fmf:41]: Please, add also `versionchanged` recording when `virtualization.hypervisor` was added to mrack plugin, similar to as https://github.com/teemtee/tmt/pull/2733/files#diff-844c430661f020411c529315cabf0d4099728fc3226a2b448ceacfc6637fe550R4

(We should have a checklist for extending HW requirement support in a plugin, we always forget something a PR needs to have...)

**Comment by skycastlelily** [comment]
updated^^

> We should have a checklist for extending HW requirement support in a
plugin, we always forget something a PR needs to have

yes, there is a checklist,will check the list, when I submit a merge
request:)

File: tmt/steps/provision/mrack.py [hunk 2]
```diff
@@ -376,6 +377,24 @@ def _transform_virtualization_is_virtualized(
     return _transform_unsupported(constraint, logger)
 
 
+def _transform_virtualization_hypervisor(
+        constraint: tmt.hardware.FlagConstraint,
```
Comment (psss) [inline tmt/steps/provision/mrack.py:381]: @happz, isn't this rather `TextConstraint`?

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for implementing this! Looks good, just one question.

File: tmt/steps/provision/mrack.py (hunk 2 above)
Comment (happz) [reply tmt/steps/provision/mrack.py:381]: Oh yeah, very much so.

Comment (psss) [reply tmt/steps/provision/mrack.py:381]: Ok, will adjust, so that we can merge this.

**Review by psss — APPROVED** [review]
I've slightly adjusted the release note in 008fc9c9.

**Review by happz — APPROVED** [review]

**Pull request merged**